File: src/MyLib.jl
module MyLib

include("../deps/deps.jl")
export delete_tms, get_tms_cstime, get_tms_cutime, get_tms_stime, get_tms_utime, libc_times, new_tms

function delete_tms(ptms)
    return ccall((:delete_tms, mylib), Void, (Ptr{Void},), ptms)
end

function get_tms_cstime(ptms)
    return ccall((:get_tms_cstime, mylib), Int32, (Ptr{Void},), ptms)
end

function get_tms_cutime(ptms)
    return ccall((:get_tms_cutime, mylib), Int32, (Ptr{Void},), ptms)
end

function get_tms_stime(ptms)
    return ccall((:get_tms_stime, mylib), Int32, (Ptr{Void},), ptms)
end

function get_tms_utime(ptms)
    return ccall((:get_tms_utime, mylib), Int32, (Ptr{Void},), ptms)
end

function libc_times(ptms)
    return ccall((:times, "libc"), Int32, (Ptr{Void},), ptms)
end

function new_tms()
    return ccall((:new_tms, mylib), Ptr{Void}, ())
end

end #module

File: deps/mylib/mylib.c
#include <stdlib.h>
#include <sys/times.h>
#include "mylib.h"

void delete_tms(struct tms *ptms) {
	free(ptms);
}

clock_t get_tms_cstime(struct tms *ptms) {
	return ptms->tms_cstime;
}

clock_t get_tms_cutime(struct tms *ptms) {
	return ptms->tms_cutime;
}

clock_t get_tms_stime(struct tms *ptms) {
	return ptms->tms_stime;
}

clock_t get_tms_utime(struct tms *ptms) {
	return ptms->tms_utime;
}

struct tms * new_tms() {
	return malloc(sizeof(struct tms));
}
